### PR TITLE
AI Image now returning. 

### LINKED ISSUE #15: Get the AI to generate a picture of the recipe.

### PR DESCRIPTION
**Pull request by samscupoftea**
1. Re-created an object called doubleResponse and added completion and imagereponse as key value pairs to Text: and Image. Much like how we did in the library yesterday 
2. Then called the object in res.json(doubleResponse);
3. Then in front end JS, I added variables for both text and image inside the .then if statement.

Screenshot 2024-04-22 at 13 43 20
Screenshot 2024-04-22 at 13 46 16

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -51,17 +51,14 @@ app.get("/openai", async (req, res) => {
     size: "1024x1024",
   });
   
-  // const image_url = imageResponse.data[0].url;
-  // console.log("Generated image URL:", image_url);
-  
 
-res.json(completion);
-res.json(imageResponse);
+const doubleResponse = {
```
Comment (samscupoftea) [inline server.js:55]: Created an object called doubleResponse to contain both text and image response. @josueJURE

File: server.js [hunk 2]
```diff
@@ -51,17 +51,14 @@ app.get("/openai", async (req, res) => {
...
+  text: completion,
+  image: imageResponse
+}
+res.json(doubleResponse);
```
Comment (samscupoftea) [inline server.js:59]: Called the object with res.json @josueJURE

File: public/index.js [hunk 3]
```diff
@@ -64,8 +64,12 @@ buttons.forEach((button) => {
       .then((response) => response.json())
       .then((data) => {
         if (resultElement) {
-          resultElement.innerHTML = `<p>${data.choices[0].message.content}</p>`;
```
Comment (samscupoftea) [inline public/index.js:67]: Deleted this as i've made the resultElement to include textcontext and imageUrl. @josueJURE

File: public/index.js [hunk 4]
```diff
@@ -64,8 +64,12 @@ buttons.forEach((button) => {
...
-          
+          const textContent = data.text.choices[0].message.content;
+          const imageUrl = data.image.data[0].url;
+          resultElement.innerHTML = `
```
Comment (samscupoftea) [inline public/index.js:69]: Created variables for text and image that can then be called via string literal to create <P> for text and  for image. @josueJURE

File: public/index.js [hunk 5]
```diff
@@ -64,8 +64,12 @@ buttons.forEach((button) => {
...
+            <p>${textContent}</p>
+            <img src="${imageUrl}" alt="Generated Image">
```
Comment (samscupoftea) [inline public/index.js:71]: It's currently not tidy or well formatted. The picture returns far too oversized. I will fix it this week. We can then look at re-locating the image to background. @josueJURE

File: server.js (hunk 1 above)
Comment (josueJURE) [reply server.js:55]: Great job! I'm just trying to understand why that didn't work when we created a similar object yesterday in the library

File: public/index.js (hunk 5 above)
Comment (josueJURE) [reply public/index.js:71]: I think I know how to relocate the image to background. I'm going to create a branch and give it a shot

File: public/index.js (hunk 3 above)
Comment (josueJURE) [reply public/index.js:67]: Yesterday in the library we were asking ourselves how to get the image generated in the backend to be displayed in the frontend. I was thinking of writing something similar to

```
  fetch("public/server.js", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(userRecipe),
    })
```

In the backend to capture the image generated by the AI and then pass it to the frontend. However, the way you've done it is way cleaner. Well done on that

File: server.js [hunk 6]
```diff
@@ -15,12 +15,18 @@ const openai = new OpenAI({
...
 });
 app.get("/openai", async (req, res) => {
+  try {
   const recipeCountryOfOrigin = req.query.recipe_country_of_origin;
   const isLactoseIntolerant = req.query.is_lactose_intolerant;
   const isVegan = req.query.is_vegan; 
 
   console.log({ isLactoseIntolerant }, { recipeCountryOfOrigin }, { isVegan });
 
+  const prompt = `Provide a recipe for a dish from ${recipeCountryOfOrigin}, taking into account the fact that the user is ${
```
Comment (josueJURE) [inline server.js:25]: Is it just me or the` prompt` variable isn't being used in the `completion `object? I can see it being used in the  `imageResponse ` object though

**Pull request merged**